# Eliminacja wartości odstających

In [55]:
load train.txt
tlab = train(:, 2:end);
means = mean(tlab)

for i=1:size(tlab,2)
sprintf("min %d for class %d is at %d", min(tlab(:, i)), i, find(tlab(:, i)==min(tlab(:, i))))
end

for i=1:size(tlab,2)
sprintf("max %d for class %d is at %d", max(tlab(:, i)), i, find(tlab(:, i)==max(tlab(:, i))))
end

means =

 Columns 1 through 6:

   1.8679e-01   1.4839e-02   2.1045e-01   2.0882e-01   7.9658e+01   1.0604e+00

 Column 7:

   9.0846e-03

ans = min 0.125 for class 1 is at 642
ans = min 0 for class 2 is at 642
ans = min 0 for class 3 is at 642
ans = min 0 for class 4 is at 642
ans = min -5.4538e-09 for class 5 is at 393
ans = min -1.214e-07 for class 6 is at 25
ans = min -3.26424e-09 for class 7 is at 539
ans = max 5.12127 for class 1 is at 186
ans = max 25.8229 for class 2 is at 186
ans = max 382.139 for class 3 is at 186
ans = max 380.858 for class 4 is at 186
ans = max 145297 for class 5 is at 186
ans = max 1934.12 for class 6 is at 186
ans = max 16.5702 for class 7 is at 186


Usuwam ze zbioru uczącego próbki 642 (wartości skrajnie małe dla większości klas) oraz 186 (wartości skrajnie duże dla większości klas)